boa/src/vm/code_block.rs

```diff
@@ -23,8 +23,8 @@ use gc::Gc;
 use std::{convert::TryInto, fmt::Write, mem::size_of};
 
 /// This represents whether a value can be read from [`CodeBlock`] code.
+/// # Safety
```

Comment:
Yeah i think @HalidOdat is needed for this one

Reply:
I will open an issue about this, so that we can merge it already. I have made the trait `pub(crate)`, so that it will at least not be shown in the public documentation and it cannot be used by external crates by mistake. #1761

Reply:
https://github.com/boa-dev/boa/issues/1761#issuecomment-1000830125